### DISCUSSION TITLE
Recent  "cannot open shared object file: Permission denied" error

### DISCUSSION THREAD

**Discussion by kxzo** [vercel/next.js#41335] [Help]
Hey, first of all I love Vercel and NextJS. Thank you all!
So, everything worked fine. In development mode, local host, everything works, as well as it worked in production mode in Vercel before.
(By the way I use yarn.)
But suddenly I get this error (for no reason), I think it is an internal Vercel server error (not caused by me):
![image](https://user-images.githubusercontent.com/66639485/195148996-3c6dd0d1-4201-4aba-8735-9db711283088.png)
And again as clear text: 
```
[POST] /api/account/signInOut
18:34:56:95
2022-10-11T16:34:57.075Z	59635609-3eb4-46da-806c-c7a534c1b593	ERROR	Error: /var/task/web/node_modules/bcrypt/lib/binding/napi-v3/bcrypt_lib.node: cannot open shared object file: Permission denied
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1189:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/var/task/web/node_modules/bcrypt/bcrypt.js:6:16)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
}
RequestId: 59635609-3eb4-46da-806c-c7a534c1b593 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

It has something to do with bcrypt..... but it is installed in yarn.

**Comment by icyJoseph** [comment]
Mmm I wonder if this is because bcrypt uses node-gyp, which, roughly speaking, allows interop with C++ libraries, IIRC. Because the lambda runs on a separate task (as you can see it is `/var/task`), rather than together with the rest of the server (a selling point of Vercel), then access to the OS might be limited? 

Could you confirm that this has started to happen with bcrypt? Like, you didn't use bcrypt before, and now that you do, this happens.

**Reply by kxzo** [reply]
I used bcrypt before and it worked. I didn't change anything in my api routes and suddenly it stopped working.
So, I have not just started recently using bcrypt, I used it a long time before and it did work.

I didn't do any weird changes, the error just weirdly occured.

**Reply by icyJoseph** [reply]
Mmm either, a redeploy changed a floating dependency of bcrypt, or there might have been some change to the way the serverless are handled. Could you open a ticket with Vercel support?

**Reply by kxzo** [reply]
Okay, I understand. Thanks for your help. I did open a ticket, and hoping for response.

**Comment by kxzo** [comment]
New update on error logs, I recently got this error in the dev console in chrome when logging in. Might help you @icyJoseph. By the way I did open a ticket...

Error object (json):
`{ "message": "Request failed with status code 500", "name": "AxiosError", "stack": "AxiosError: Request failed with status code 500\n    at https://shiromeet.vercel.app/_next/static/chunks/127-c45a683fc25adf5f.js:127:148235\n    at XMLHttpRequest.u (https://shiromeet.vercel.app/_next/static/chunks/127-c45a683fc25adf5f.js:127:148380)", "config": { "transitional": { "silentJSONParsing": true, "forcedJSONParsing": true, "clarifyTimeoutError": false }, "transformRequest": [ null ], "transformResponse": [ null ], "timeout": 0, "xsrfCookieName": "XSRF-TOKEN", "xsrfHeaderName": "X-XSRF-TOKEN", "maxContentLength": -1, "maxBodyLength": -1, "env": {}, "headers": { "Accept": "application/json, text/plain, */*", "Content-Type": "application/json" }, "method": "post", "url": "https://shiromeet.vercel.app/api/account/signInOut", "data": "{\"email\":\"test.test@test.com\",\"password\":\"test\",\"token\":\"\"}" }, "code": "ERR_BAD_RESPONSE", "status": 500 }`

**Reply by icyJoseph** [reply]
mm nope, I guess this is the result of trying to unwrap the failed response from the API route.
